recommender-system/README.md

```diff
@@ -1,7 +1,87 @@
+- The messages that were exchanged in the different conversations he was involved in (for a limited amount of time).
+
+# Structure of the Machine Learning Algorithm:
+# Actual-Online sequence diagrams
+## To get a recommendation list
+```mermaid
+sequenceDiagram
+      actor User as User
+      participant MM as Master Model
+      participant Model2 as User Model
+      participant m as Mapping
+      User ->> MM: GET Recommendation for user u (HTTP API Gateway)
+      Note right of User: API Gateway HTTP - userid in the path
+      MM ->> m: GET mapping
+      alt NO existing mapping
+      m ->> MM: Return Error (No existing mapping)
+      Note right of MM: TO DO : Avoid the HTTP error by creating the mapping
+      end
+
+      alt Existing mapping
+      m ->> MM: Return mapped userid
+      MM ->> MM: Process & Check HTTP request
+      MM ->> Model2: GET User gradient of Master Model and Recommendation
+      Note right of MM: Call between lambdas on AWS
+      Model2 ->> Model2: Process and update User Model
+      Model2 ->> MM: Recommendation and gradient transmission
+      MM ->> User: Recommendation transmission
+      MM ->> MM: Master Model update
+      end
+```
+
+## To update the rating matrix
+```mermaid
+sequenceDiagram
+      actor User as User
+      participant R as Ratings
+      participant m as Mapping
+      User ->> User: Has participated to a conversation
+      User ->> R: UPDATE Ratings for userid
+      Note right of User: API Gateway HTTP
+      R ->> m: GET mapping (Supposed to exist bcs required to enter a conversation)
+      alt NO existing mapping
+      m ->> R: RETURN Error (No existing mapping)
+      Note right of R: Error could happen if *userid* change during a conversation
+      end
+
+      alt Existing mapping
+      m ->> R: RETURN mapped userid
+      R ->> R: PROCESS & CHECK HTTP request
+      R ->> R: Row userid UPDATE of R matrix
+      R ->> User: RETURN 200 'Update succeded'
+      end
+```
+# Future-Local sequence diagrams
+## Desired-Local and Privacy preserving sequence diagram
+```mermaid
+sequenceDiagram
+      actor User as User - Model
+      participant Model as Master Model
+      User ->> Model: GET Master Model
+      Note right of User: API Gateway HTTP
+      Model ->> User: Response: Master Model matrix
+      User ->> User: Reco., gradient computation, update User Model
+      User ->> Model: POST gradient Master Model matrix
+      Model ->> User: Response : StatusCode: 200
+      Model ->> Model: Master Model update
+```
+
+
+
 # AWS Setup:
-We want to use Python and the NumPy module in an AWS Lambda function. NumPy is not available in the Python AWS default envrionment. Hence, the only working solution found was to create a dockerized environnement.
+We use Python and the NumPy module in an AWS Lambda function. NumPy is not available in the Python AWS default environment. Hence, the only working solution found was to create a dockerized environnement.
```

Comment:
Hence, the only working solution found was to create a dockerized environnement. > c'est pas vrai 😇, c'est la solution qu'on a choisi